gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy

```diff
@@ -0,0 +1,531 @@
+        // in people's codebases
+
+        when:
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                @SuppressWarnings("for-rollout:ArrayToString")
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        then:
+        runTasksSuccessfully('compileJava')
+    }
+
+    def 'ensure error prone checks are disabled in generated code'() {
+        // language=Java
+        def erroringCode = '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+
+        when:
+        def sourceSet1 = new File(sourceSetRoot, '/src/generated')
+        def sourceSet2 = new File(sourceSetRoot, '/build/somePlace')
+
+        writeJavaSourceFile(erroringCode, sourceSet1)
+        writeJavaSourceFile(erroringCode.replace('App', 'App2'), sourceSet2)
+
+        buildFile << """
+            sourceSets.main.java.srcDirs('${projectDir.relativePath(sourceSet1)}', '${projectDir.relativePath(sourceSet2)}')
+        """.stripIndent(true)
+
+        then:
+        runTasksSuccessfully('compileJava')
+    }
+
+    def 'can apply patches for a check if added to the patchChecks list'() {
+        // language=Gradle
+        buildFile << '''
+            suppressibleErrorProne {
+                patchChecks.add('ArrayToString')
+            }
+        '''.stripIndent(true)
+
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        runTasksSuccessfully('compileJava', '-PerrorProneApply')
+
+        then:
+        runTasksSuccessfully('compileJava')
+
+        appJava.text.contains('Arrays.toString(new int[3])')
+    }
+
+    def 'does not apply patches for a check if not added to the patchChecks list'() {
+        // language=Gradle
+        buildFile << '''
+            suppressibleErrorProne {
+                // To make sure set is not empty
+                patchChecks = ['SomeCheck']
+            }
+        '''.stripIndent(true)
+
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        runTasksSuccessfully('compileJava', '-PerrorProneApply')
```

Comment:
why was this successful? shouldn't it fail?

Reply:
or is apply just fix any thing that's fixable and leave the rest alone?

Reply:
Apply just applies fixes, it doesn't fail if there are errors left over.